2.4.0-RC2 BoneCP exception: No configuration setting found for key 'defaultCatalog'

Changing jdbc connection pool back to BoneCP with `play.db.pool = bonecp` setting as suggested by the migration guide doesn't work properly right now.

```
play.api.Configuration$$anon$1: Configuration error[Cannot connect to database [default]]
	at play.api.Configuration$.configError(Configuration.scala:178) ~[play_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at play.api.Configuration.reportError(Configuration.scala:829) ~[play_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at play.api.db.DefaultDBApi$$anonfun$connect$1.apply(DBApi.scala:74) ~[play-jdbc_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at play.api.db.DefaultDBApi$$anonfun$connect$1.apply(DBApi.scala:68) ~[play-jdbc_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at scala.collection.immutable.List.foreach(List.scala:381) ~[scala-library-2.11.6.jar:na]
	at play.api.db.DefaultDBApi.connect(DBApi.scala:68) ~[play-jdbc_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at play.api.db.DBApiProvider.get$lzycompute(DBModule.scala:72) ~[play-jdbc_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at play.api.db.DBApiProvider.get(DBModule.scala:62) ~[play-jdbc_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at play.api.db.DBApiProvider.get(DBModule.scala:58) ~[play-jdbc_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81) ~[guice-4.0.jar:na]
Caused by: com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'defaultCatalog'
	at com.typesafe.config.impl.SimpleConfig.findKeyOrNull(SimpleConfig.java:152) ~[config-1.3.0-M3.jar:na]
	at com.typesafe.config.impl.SimpleConfig.findOrNull(SimpleConfig.java:170) ~[config-1.3.0-M3.jar:na]
	at com.typesafe.config.impl.SimpleConfig.findOrNull(SimpleConfig.java:193) ~[config-1.3.0-M3.jar:na]
	at com.typesafe.config.impl.SimpleConfig.findOrNull(SimpleConfig.java:198) ~[config-1.3.0-M3.jar:na]
	at com.typesafe.config.impl.SimpleConfig.getIsNull(SimpleConfig.java:208) ~[config-1.3.0-M3.jar:na]
	at play.api.PlayConfig.getOptional(Configuration.scala:951) ~[play_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at play.api.PlayConfig.getOptionalDeprecated(Configuration.scala:996) ~[play_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at play.api.db.BoneConnectionPool.create(BoneCPModule.scala:67) ~[play-jdbc_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at play.api.db.PooledDatabase.createDataSource(Database.scala:276) ~[play-jdbc_2.11-2.4.0-RC2.jar:2.4.0-RC2]
	at play.api.db.DefaultDatabase.dataSource$lzycompute(Database.scala:201) ~[play-jdbc_2.11-2.4.0-RC2.jar:2.4.0-RC2]
```

--- a/framework/src/play-jdbc/src/main/scala/play/api/db/BoneCPModule.scala
+++ b/framework/src/play-jdbc/src/main/scala/play/api/db/BoneCPModule.scala
@@ -64,7 +64,7 @@ class BoneConnectionPool @Inject() (environment: Environment) extends Connection
       case unknown => throw config.reportError("bonecp.isolation",
         s"Unknown isolation level [$unknown]")
     }
-    val catalog = config.getOptionalDeprecated[String]("defaultCatalog", "bonecp.defaultCatalog")
+    val catalog = config.getOptionalDeprecated[String]("bonecp.defaultCatalog", "defaultCatalog")
     val readOnly = config.getDeprecated[Boolean]("bonecp.readOnly", "readOnly")
 
     datasource.setClassLoader(environment.classLoader)

--- a/framework/src/play-jdbc/src/test/scala/play/api/db/ConnectionPoolConfigSpec.scala
+++ b/framework/src/play-jdbc/src/test/scala/play/api/db/ConnectionPoolConfigSpec.scala
@@ -1,0 +1,70 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package play.api.db
+
+import javax.inject.Inject
+import play.api.test._
+
+class ConnectionPoolConfigSpec extends PlaySpecification {
+
+  "DBModule bindings" should {
+
+    "use HikariCP when default pool is default" in new WithApplication(FakeApplication(
+      additionalConfiguration = Map(
+        "db.default.url" -> "jdbc:h2:mem:default",
+        "db.other.driver" -> "org.h2.Driver",
+        "db.other.url" -> "jdbc:h2:mem:other"
+      )
+    )) {
+      val db = app.injector.instanceOf[DBApi]
+      db.database("default").withConnection { c =>
+        c.getClass.getName must contain("hikari")
+      }
+    }
+
+    "use HikariCP when default pool is 'hikaricp'" in new WithApplication(FakeApplication(
+      additionalConfiguration = Map(
+        "play.db.pool" -> "hikaricp",
+        "db.default.url" -> "jdbc:h2:mem:default",
+        "db.other.driver" -> "org.h2.Driver",
+        "db.other.url" -> "jdbc:h2:mem:other"
+      )
+    )) {
+      val db = app.injector.instanceOf[DBApi]
+      db.database("default").withConnection { c =>
+        c.getClass.getName must contain("hikari")
+      }
+    }
+
+    "use BoneCP when default pool is 'bonecp'" in new WithApplication(FakeApplication(
+      additionalConfiguration = Map(
+        "play.db.pool" -> "bonecp",
+        "db.default.url" -> "jdbc:h2:mem:default",
+        "db.other.driver" -> "org.h2.Driver",
+        "db.other.url" -> "jdbc:h2:mem:other"
+      )
+    )) {
+      val db = app.injector.instanceOf[DBApi]
+      db.database("default").withConnection { c =>
+        c.getClass.getName must contain("bonecp")
+      }
+    }
+
+    "use BoneCP when database-specific pool is 'bonecp'" in new WithApplication(FakeApplication(
+      additionalConfiguration = Map(
+        "db.default.pool" -> "bonecp",
+        "db.default.url" -> "jdbc:h2:mem:default",
+        "db.other.driver" -> "org.h2.Driver",
+        "db.other.url" -> "jdbc:h2:mem:other"
+      )
+    )) {
+      val db = app.injector.instanceOf[DBApi]
+      db.database("default").withConnection { c =>
+        c.getClass.getName must contain("bonecp")
+      }
+    }
+
+  }
+
+}

